src/functions.php

```diff
@@ -173,37 +199,40 @@ function uploadImages(array $vcards, array $config, array $phonebook, callable $
 /**
  * Dissolve the groups of iCloud contacts
  *
- * @param stdClass[] $vcards
- * @return stdClass[]
+ * @param mixed[] $vcards
+ * @return mixed[]
  */
 function dissolveGroups(array $vcards): array
 {
     $groups = [];
 
     // separate iCloud groups
+    /** @var \stdClass $vcard */
     foreach ($vcards as $key => $vcard) {
-        if (isset($vcard->xabsmember)) {
-            if (array_key_exists($vcard->fullname, $groups)) {
-                $groups[$vcard->fullname] = array_merge($groups[$vcard->fullname], $vcard->xabsmember);
-            } else {
-                $groups[$vcard->fullname] = $vcard->xabsmember;
+        if (isset($vcard->{'X-ADDRESSBOOKSERVER-KIND'})) {
+            if ($vcard->{'X-ADDRESSBOOKSERVER-KIND'} == 'group') {      // identifier
+                foreach ($vcard->{'X-ADDRESSBOOKSERVER-MEMBER'} as $member) {
+                    $member = str_replace(['urn:', 'uuid:'], ['', ''], (string)$member);
```

Comment:
iCloud use both prefix - but I´m not sure if this is fix - so I decided to replace them separatly